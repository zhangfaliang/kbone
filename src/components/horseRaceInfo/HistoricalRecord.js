import React from 'react';
import styles from './index.less';

const HistoricalRecord = ({ historyRecord = '', prefixCls = 'single' }) => {
  const clsStr = styles[`${prefixCls}-horse-historical-record`];
  return <div className={clsStr}>{historyRecord}</div>;
};

export default HistoricalRecord;
