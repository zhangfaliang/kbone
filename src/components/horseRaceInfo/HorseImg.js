import React, { createRef } from 'react';
import styles from './index.less';
const defaultHorseTeam = require('../../assets/horse/defaultHorseTeam.png');

class HorseImg extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      imgUrl: props.picUrl || props.defaultUrl,
    };
  }
  static getDerivedStateFromProps(props, state) {
    if (props.imgUrl !== state.imgUrl) {
      return { activeKey: props.imgUrl };
    }
    return null;
  }
  onError = () => {
    this.setState({
      imgUrl: this.props.defaultUrl,
    });
  };
  render() {
    const { picUrl, prefixCls } = this.props;
    const clsStr = styles[`${prefixCls}-horse-img`];
    return <img className={clsStr} src={this.state.imgUrl} alt="" onError={this.onError} />;
  }
}
HorseImg.defaultProps = {
  defaultUrl: defaultHorseTeam,
  prefixCls: 'single',
  picUrl: '',
};
export default HorseImg;
