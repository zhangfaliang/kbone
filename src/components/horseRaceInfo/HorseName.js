import React from 'react';
import styles from './index.less';

//todo 马名：黑色字体为马名，最多显示22位，超过部分用“..."显示。
const HorseName = ({ shortName = '', prefixCls = '' }) => {
  const clsStr = styles[`${prefixCls}-horse-name`];
  return <div className={clsStr}>{shortName}</div>;
};

export default HorseName;
