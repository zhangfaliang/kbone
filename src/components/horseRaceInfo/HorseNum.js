import React from 'react';
import styles from './index.less';

const HorseNum = ({ horseCode = '', prefixCls = '' }) => {
  const clsStr = styles[`${prefixCls}-horse-num`];
  return <div className={clsStr}>{horseCode}</div>;
};

export default HorseNum;
