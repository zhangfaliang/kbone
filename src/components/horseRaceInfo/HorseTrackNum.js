import React from 'react';
import styles from './index.less';

const HorseTrackNum = ({ horseTrack = '', prefixCls = 'single' }) => {
  const clsStr = styles[`${prefixCls}-horse-track-num`];
  return <div className={clsStr}>{horseTrack}</div>;
};
export default HorseTrackNum;
