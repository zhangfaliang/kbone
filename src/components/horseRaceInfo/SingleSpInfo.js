import React from 'react';
import styles from './index.less';

const SPInfo = ({ textInfo = '', prefixCls = 'default' }) => {
  const clsStr = styles[`${prefixCls}-sp-info`];
  return <div className={clsStr}>{textInfo}</div>;
};

SPInfo.COMPONENT_NAME = 'SPINFO';
export default SPInfo;
