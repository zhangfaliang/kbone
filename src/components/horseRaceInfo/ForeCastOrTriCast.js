import React from 'react';
import styles from './index.less';
import HorseNum from './HorseNum';
import HorseName from './HorseName';

const ForeCastOrTriCast = ({ horseCode = '', shortName = '', prefixCls = 'fore-or-tri' }) => {
  const clsStr = styles[`${prefixCls}-horse-info`];
  return (
    <div className={clsStr}>
      <HorseNum prefixCls={prefixCls} horseCode={horseCode} />
      <HorseName prefixCls={prefixCls} shortName={shortName} />
    </div>
  );
};

ForeCastOrTriCast.COMPONENT_NAME = 'FORECASTORTRICAST';
export default ForeCastOrTriCast;
