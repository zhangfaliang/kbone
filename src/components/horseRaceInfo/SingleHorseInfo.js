import React from 'react';
import HorseImg from './HorseImg';
import HorseNum from './HorseNum';
import HorseTrackNum from './HorseTrackNum';
import HistoricalRecord from './HistoricalRecord';
import HorseName from './HorseName';
import styles from './index.less';

class SingleHorseInfo extends React.Component {
  static COMPONENT_NAME = 'SINGLEHORSERACEINFO';

  render() {
    const {
      historyRecord,
      horseCode,
      horseTrack,
      picUrl,
      shortName,
      horseTrainer,
      jockey,
      horseAge,
      jockeyWeight,
    } = this.props;
    const LeftInfo = (
      <div className={styles.leftInfo}>
        <div className={styles.top}>
          <div className={styles.left}>
            <HorseNum prefixCls="single" horseCode={horseCode} />
            <HorseTrackNum prefixCls="single" horseTrack={horseTrack} />
          </div>
          <HorseImg prefixCls="single" picUrl={picUrl} />
        </div>
        <HistoricalRecord prefixCls="single" historyRecord={historyRecord} />
      </div>
    );

    const rightInfo = (
      <div className={styles.rightInfo}>
        <HorseName prefixCls="single" shortName={shortName} />
        {/*//todo 右侧是骑手名字。每个名字最多展示12位，超过12位用“..."显示*/}
        <div className={styles.horseman}>
          <span>{horseTrainer}</span> <span>/ </span>
          <span>{jockey}</span>
        </div>
        <div>
          <span className={styles.age}> {horseAge}</span>
          <span className={styles.weight}>{jockeyWeight}</span>
        </div>
      </div>
    );

    return (
      <div className={styles['single-horse-info']}>
        {LeftInfo}
        {rightInfo}
      </div>
    );
  }
}

SingleHorseInfo.defaultProps = {
  horseCode: '', //马号
  picUrl: '', // 图片
  horseTrack: '', //赛道
  shortName: '', //马名
  horseTrainer: '', //训练师
  jockey: '', //骑手
  historySp: '', //历史赔率
  horseAge: '',
  jockeyWeight: 100,
};
export default SingleHorseInfo;
