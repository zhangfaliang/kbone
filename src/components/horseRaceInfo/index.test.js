import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import ForeCastOrTriCast from './ForeCastOrTriCast';
import HistoricalRecord from './HistoricalRecord';
import HorseImg from './HorseImg';
import HorseName from './HorseName';
import HorseNum from './HorseNum';
import HorseTrackNum from './HorseTrackNum';
import SingleHorseInfo from './SingleHorseInfo';
import SingleSpInfo from './SingleSpInfo';

describe('测试 horseRaceInfo', async () => {
  it('ForeCastOrTriCast', () => {
    const wrapper = render(<ForeCastOrTriCast horseCode="wer" shortName="wer" prefixCls=" " />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HistoricalRecord', () => {
    const wrapper = render(<HistoricalRecord />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HorseImg', () => {
    const wrapper = render(<HorseImg />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HorseImg', () => {
    const wrapper = mount(<HorseImg imgUrl="234" />);
    wrapper.setProps({
      imgUrl: '23',
    });
    wrapper.setState({
      imgUrl: '23',
    });
    expect('23').toBe(wrapper.prop('imgUrl'));
  });
  it('HorseImg', () => {
    const wrapper = mount(<HorseImg imgUrl="23" />);
    wrapper.setProps({
      imgUrl: '23',
    });
    expect('23').toBe(wrapper.prop('imgUrl'));
  });
  it('HorseName', () => {
    const wrapper = render(<HorseName />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('HorseNum', () => {
    const wrapper = render(<HorseNum />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('HorseTrackNum', () => {
    const wrapper = render(<HorseTrackNum horseTrack="wer" prefixCls="single1" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('SingleHorseInfo', () => {
    const wrapper = render(<SingleHorseInfo />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('SingleSpInfo', () => {
    const wrapper = render(<SingleSpInfo />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
