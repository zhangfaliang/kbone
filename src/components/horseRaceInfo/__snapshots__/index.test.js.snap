// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 horseRaceInfo ForeCastOrTriCast 1`] = `
<div
  class=" -horse-info"
>
  <div
    class=" -horse-num"
  >
    wer
  </div>
  <div
    class=" -horse-name"
  >
    wer
  </div>
</div>
`;

exports[`测试 horseRaceInfo HistoricalRecord 1`] = `
<div
  class="single-horse-historical-record"
/>
`;

exports[`测试 horseRaceInfo HorseImg 1`] = `
<img
  alt=""
  class="single-horse-img"
  src="[object Object]"
/>
`;

exports[`测试 horseRaceInfo HorseName 1`] = `
<div
  class="-horse-name"
/>
`;

exports[`测试 horseRaceInfo HorseNum 1`] = `
<div
  class="-horse-num"
/>
`;

exports[`测试 horseRaceInfo HorseTrackNum 1`] = `
<div
  class="single1-horse-track-num"
>
  wer
</div>
`;

exports[`测试 horseRaceInfo SingleHorseInfo 1`] = `
<div
  class="single-horse-info"
>
  <div
    class="leftInfo"
  >
    <div
      class="top"
    >
      <div
        class="left"
      >
        <div
          class="single-horse-num"
        />
        <div
          class="single-horse-track-num"
        />
      </div>
      <img
        alt=""
        class="single-horse-img"
        src="[object Object]"
      />
    </div>
    <div
      class="single-horse-historical-record"
    />
  </div>
  <div
    class="rightInfo"
  >
    <div
      class="single-horse-name"
    />
    <div
      class="horseman"
    >
      <span />
       
      <span>
        / 
      </span>
      <span />
    </div>
    <div>
      <span
        class="age"
      >
         
      </span>
      <span
        class="weight"
      >
        100
      </span>
    </div>
  </div>
</div>
`;

exports[`测试 horseRaceInfo SingleSpInfo 1`] = `
<div
  class="default-sp-info"
/>
`;
